### ISSUE TITLE
Blockly.Procedures.isNameUsed returning wrong values

### ISSUE THREAD

**Issue by smaameri** [LLK/scratch-blocks#1175]
The Blockly.Procedures.isNameUsed function is returing the opposite of what it says it should. 

As per the definition it says:

`//  @return {boolean} True if the name is used, otherwise return false`

however the code is returning opposite. Compared with the Blockly code, the return true/false locations are also switched. I.e the function is returning true when it should return false, and false when it should return true. Any reasons for this?

```
/**
 * Return if the given name is already a procedure name.
 * @param {string} name The questionable name.
 * @param {!Blockly.Workspace} workspace The workspace to scan for collisions.
 * @param {Blockly.Block=} opt_exclude Optional block to exclude from
 *     comparisons (one doesn't want to collide with oneself).
 * @return {boolean} True if the name is used, otherwise return false.
 */

Blockly.Procedures.isNameUsed = function(name, workspace, opt_exclude) {
  var blocks = workspace.getAllBlocks();
  // Iterate through every block and check the name.
  for (var i = 0; i < blocks.length; i++) {
    if (blocks[i] == opt_exclude) {
      continue;
    }
    if (blocks[i].getProcedureDef) {
      var procName = blocks[i].getProcedureDef();
      if (Blockly.Names.equals(procName[0], name)) {
        return false;
      }
    }
  }
  return true;
};

```

**Comment by thisandagain** [comment]
@rachel-fenichel @paulkaplan Any thoughts on this?

**Comment by rachel-fenichel** [comment]
Yup, that looks wrong!  Fixed in blockly [here](https://github.com/google/blockly/commit/234c53157f676978ab9de232ed28305720443003), with a commit comment of "Fix Blockly.Procedures.isNameUsed return values.
Add tests for Procedures.isNameUsed() so this bug never happens again."

However, fixing that won't fix `isNameUsed` right now--it's also calling `getProcedureDef` to check whether something is a procedure.  Since that's not defined on any blocks, it'll just always return true.

I'm working on implementing procedures right now, so I'll add this to my list.

**Comment by rachel-fenichel** [comment]
However, this may be an argument for calling [`getProcCode`](https://github.com/LLK/scratch-blocks/blob/develop/blocks_vertical/procedures.js#L157) `getProcedureDef` instead, to keep compatibility.  Something to think about.

**Comment by rachel-fenichel** [comment]
I'm going to leave this open and assigned to me, because it's a real issue with the code.

**Comment by cwillisf** [comment]
scratch-blocks v2 is a complete rewrite built on top of upstream [Blockly](https://github.com/RaspberryPiFoundation/blockly/). The internal code referenced in this issue no longer exists. Thank you for reporting it!

If you believe this issue is still relevant to scratch-blocks v2, please let us know and we can reopen it.
